Enabled the "param" option in the "run" command

bisenpiyush opened the pull request:
## Changes proposed in this pull request
While fixing SNAP-3006, "param" option has been disabled for syncup with snappydata document, which turns into the cause of Hydra test fail. Enabled this option again.
## Patch testing
Tested with automated test
## Is precheckin with -Pstore clean?
Yet to run
## ReleaseNotes changes
NA
## Other PRs 
NA

Commit:
Enabled the "param" option

sumwale commented on gemfirexd/tools/src/main/java/com/pivotal/gemfirexd/tools/internal/MiscTools.java:
```diff
@@ -120,7 +120,7 @@ public final void addCommandOptions(final Options opts) {
           addEncodingOption(opts);
           addIgnoreErrorsOption(opts);
-//          addParamOption(opts);
+          addParamOption(opts);
```
Why has addNumTimesToRun been removed?

Also refactor the addConnectionOptions in ToolsBase to split out the ones used here (without locators/mcast options) into a separate method so that override here can call that method instead of duplicating the code.

bisenpiyush replied:
As SNAP-3006 asked to sync up "run" command's "usage" with Snappydata docs, removed addNumTimesToRun. 
Now reverted back the changes, and only "-locators" has been hide from "usage". For that as suggested split out addConnectionOptions into a separate method addOptionalConnectionOptions.

sumwale reviewed (changes requested):
see comment about refactoring

Commit:
Modified as per review comment. Split addConnectionOption method

sumwale commented on gemfirexd/tools/src/main/java/com/pivotal/gemfirexd/tools/internal/ToolsBase.java:
```diff
@@ -269,12 +270,6 @@ protected void addConnectionOptions(final Options opts) {
-        .withDescription(LocalizedResource.getMessage(
-            "TOOLS_LOCATORS_MESSAGE")).create(LOCATORS);
-    opts.addOption(opt);
-
```
also move TOOLS_MCAST_PORT_MESSAGE, TOOLS_MCAST_ADDRESS_MESSAGE, TOOLS_BIND_ADDRESS_MESSAGE and TOOLS_AUTH_PROVIDER_MESSAGE into additional options since these are only for embedded mode connections which are not supported anymore

bisenpiyush replied:
Moved

Commit:
Moved TOOLS_MCAST_PORT_MESSAGE,
TOOLS_MCAST_ADDRESS_MESSAGE, TOOLS_BIND_ADDRESS_MESSAGE and TOOLS_AUTH_PROVIDER_MESSAGE
 into additional options

sumwale reviewed (approved):
LGTM

Commit:
Merge branch 'snappy/master' into SNAP-3105
